### ISSUE TITLE
Phoromatic: Test result aggregation

### ISSUE THREAD

**Issue by swarth100** [phoronix-test-suite/phoronix-test-suite#263]
[Not an actual issue, borderline question/feature request]

When comparing results on Phoromatic, I was wondering if there was a way to export the aggregated results as a new standalone test result. 

Exporting it in this way would then allow me to hack into the XML removing redundant results, or removing tests that didn't really run correctly on all machines.

Should we assume we have a **result1** and **result2** file, A *similar* result can be achieved by running a dummy fast benchmark, and replacing the contents of the results file with the aggregated data. This though isn't really nice nor scaleable.

I know that comparison links can be shared, though the elements present within this comparison cannot be clearly customised (if not by changing the XML of all the test results contributing to the comparisons file).

Is this feature present / would there be an easier way to achieve a similar result?

**Comment by michaellarabel** [comment]
You should be able to achieve this functionality already. Quick steps used:

Go to the results page, select the results to compare, and click 'Compare'

You should then be on the result page with the two or more results to be shown, e.g. http://192.168.1.110:8888/?result/5799to643x1k,4018rhyyjeyosko8 

From the right hand side should be a "Download As XML" link. If you click there a new page should open showing the dynamically merged XML data that you can then download and edit locally.

Let me know if this doesn't work out or you uncover an issue in the process. Thanks.

**Comment by michaellarabel** [comment]
Oh and  the format of the URL in case you don't happen to see the link: http://192.168.1.110:8888/?result/5799to643x1k,4018rhyyjeyosko8/&download=xml

Basically appending a /&download=xml to your result URL will produce the raw XML rather than rendering the web page.

**Comment by swarth100** [comment]
Oh, thanks for the clarification!

I do confirm that I am able to download the raw XML, and hence customise it in the aggregated form (which does simplify the procedure greatly).

My question would be if there is any way to upload a customised XML saving it as an actual result within phoromatic. Or more clearly:

PROBLEM:
I would like to store custom aggregated/exported results from within the Phoromatic Results page.
(How does this differ from simply providing a comparison link of the form http://server/?result/file1,file2 ?
There might be a number of results present within file1, or file2 which I feel are unimportant within the aggregated view. Storing custom tests would also allow me to merge results of local runs executed without Phoromatic, or of saving locally openbenchmarking.org results)

CURRENT APPROACH:
I tried manually injecting result files within the results directory on the Phoromatic Server
`/var/lib/phoronix-test-suite/phoromatic/accounts/ABCDEF/results`
Though as far as I understand the code there is a sqllite middlelayer that handles the result file name/result file location. This meant I could not add custom results, even though I could customise existing result files)

Hence my CURRENT SOLUTION:
I currently work around this issue by creating a fake placeholder test, which will be added to the database and will have a results file stored on the server. I then proceed to edit the placeholder test with the custom xml data
